mssql/compiler.py

```diff
@@ -470,11 +480,36 @@ def as_sql(self):
 
         placeholder_rows, param_rows = self.assemble_as_sql(fields, value_rows)
 
-        if self.get_returned_fields() and self.connection.features.can_return_id_from_insert:
-            result.insert(0, 'SET NOCOUNT ON')
-            result.append((values_format + ';') % ', '.join(placeholder_rows[0]))
-            params = [param_rows[0]]
-            result.append('SELECT CAST(SCOPE_IDENTITY() AS bigint)')
+        if self.get_returned_fields() and self.can_return_columns_from_insert():
+            if self.can_return_rows_from_bulk_insert():
+                if not(self.query.fields):
+                    # There isn't really a single statement to bulk multiple DEFAULT VALUES insertions,
+                    # so we have to use a workaround:
+                    # https://dba.stackexchange.com/questions/254771/insert-multiple-rows-into-a-table-with-only-an-identity-column
+                    result = ['MERGE INTO %s' % qn(opts.db_table)]
+                    result.append("""
+                        USING (SELECT TOP %s * FROM master..spt_values) T
```

Comment:
To be convincing regarding the viability of that workaround, it would be nice to "prove" that this system table always existsand is always loaded with more rows than the max objs number in a bulk insert (the link above by itself doesn't go that far I believe)